azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/LoadBalancerImpl.java

```diff
@@ -401,9 +431,40 @@ public ProbeImpl defineHttpProbe(String name) {
         return new ProbeImpl(name, inner, this);
     }
 
+    @Override
+    public LoadBalancerImpl withoutProbe(String name) {
+        if (this.httpProbes.containsKey(name)) {
+            this.httpProbes.remove(name);
+        } else if (this.tcpProbes.containsKey(name)) {
+            this.tcpProbes.remove(name);
+        }
+
+        List<ProbeInner> probes = this.inner().probes();
+        if (probes != null) {
+            for (int i = 0; i < probes.size(); i++) {
+                if (probes.get(i).name().equalsIgnoreCase(name)) {
+                    probes.remove(i);
+                    break;
+                }
+            }
+        }
+
+        return this;
```

Comment:
No, the probe names are unique. Inner LB only has one collection of probes. But the Swagger-based definition has issues because different properties are applicable depending on the protocol of the probe. So a more proper way to model this is to have two different types for HTTP and TCP probes, deriving from a common base, which is what I've done here.
But I kept removing them simple - since it's just one collection at the end of the day and the names are unique --- hence no distinction between removing HTTP probe and TCP probe.